integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala

```diff
@@ -145,6 +152,208 @@ object CarbonStore {
+              load.getSegmentStatus.getMessage,
+              startTime,
+              endTime,
+              mergedTo,
+              load.getFileFormat.toString,
+              load.getVisibility,
+              Strings.formatSize(dataSize.toFloat),
+              Strings.formatSize(indexSize.toFloat),
+              isSorted,
+              sortColumns)
+          } else {
+            Row(
+              load.getLoadName,
+              load.getSegmentStatus.getMessage,
+              startTime,
+              endTime,
+              mergedTo,
+              load.getFileFormat.toString,
+              Strings.formatSize(dataSize.toFloat),
+              Strings.formatSize(indexSize.toFloat),
+              isSorted,
+              sortColumns)
+          }
+        }.toSeq
+    } else {
+      Seq.empty
+    }
+  }
+
+  private def getSortColumnsOfSegment(
+      load: LoadMetadataDetails,
+      readCommitScope: ReadCommittedScope,
+      tableDataMap: TableDataMap,
+      hadoopConf: Configuration
+  ): (String, String) = {
+    var isSorted = ""
+    var sortColumns = ""
+    if (load.getFileFormat == FileFormat.ROW_V1) {
+      isSorted = "false"
+    } else if (load.getVisibility.equalsIgnoreCase("true")) {
+      val segment = new Segment(load.getLoadName(), load.getSegmentFile(), readCommitScope)
+      val dataMapFactory = tableDataMap.getDataMapFactory().asInstanceOf[BlockletDataMapFactory]
+      val indexIdents = dataMapFactory.getTableBlockIndexUniqueIdentifiers(segment)
+      val indexIterator = indexIdents.iterator()
+      if (indexIterator.hasNext) {
+        val indexIdent = indexIterator.next()
+        var indexHeader: IndexHeader = null
+        var indexContent: Array[Byte] = null
+        val indexFilePath = indexIdent.getIndexFilePath + CarbonCommonConstants.FILE_SEPARATOR +
+                        indexIdent.getIndexFileName
+        if (indexIdent.getMergeIndexFileName != null) {
+          val indexFileStore = new SegmentIndexFileStore(hadoopConf)
+          try {
+            indexFileStore.readMergeFile(indexFilePath)
+          } catch {
+            case ex: IOException =>
+              LOGGER.error(ex)
+          }
+          val iterator = indexFileStore.getCarbonIndexMap.entrySet().iterator()
+          if (iterator.hasNext) {
+            indexContent = iterator.next().getValue
+          }
+        }
+        val indexReader = new CarbonIndexFileReader()
+        try {
+          if (indexContent == null) {
+            indexReader.openThriftReader(indexFilePath)
+          } else {
+            indexReader.openThriftReader(indexContent)
+          }
+          // get the index header
+          indexHeader = indexReader.readIndexHeader()
+        } catch {
+          case ex: IOException =>
+            LOGGER.error(ex)
+        } finally {
+          indexReader.closeThriftReader()
+        }
+        if (indexHeader != null && indexHeader.isSetIs_sort) {
+          if (indexHeader.is_sort) {
```

Comment:
```
isSorted = indexHeader.is_sort
```
if `isSorted` is false and the `indexHeader.getTable_columns` returns empty list, then we do not need the if-else here

Reply:
default is an empty string for not set is_sort cases.
when it set is_sort, it should be false.